src/Jobs/DeleteProducts.php

```diff
@@ -37,6 +37,11 @@ public function get_name(): string {
 	 * @throws ProductSyncerException If an error occurs. The exception will be logged by ActionScheduler.
 	 */
 	public function process_items( array $product_id_map ) {
+		$ready_ids = $this->product_repository->find_delete_product_ids( $product_id_map );
```

Comment:
`find_delete_product_ids` method returns product ids for deletion using product ids for deletion. So it does filtering, but inside a Repository type of class, which should, to my opinion, not incorporate filtering and fetch all it finds from the storage by the query it produces. We could 'glue' Repositories and Filters inside Jobs classes.

Some additional thoughts on Repository filtering are here as well: https://github.com/woocommerce/google-listings-and-ads/pull/1238/files#r810348456